dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/checkpoint/CheckpointKafkaSaslPlainTest.java

```diff
@@ -0,0 +1,197 @@
+
+    @Server(SERVER_NAME)
+    @TestServlet(contextRoot = APP_NAME, servlet = KafkaSaslTestServlet.class)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+
+        ConnectorProperties outgoingProperties = simpleOutgoingChannel(null, BasicMessagingBean.CHANNEL_OUT);
+
+        ConnectorProperties incomingProperties = simpleIncomingChannel(null, BasicMessagingBean.CHANNEL_IN, APP_GROUP_ID);
+
+        ConnectorProperties connectorProperties = new ConnectorProperties(Direction.CONNECTOR, "liberty-kafka")
+                        .addAll(SaslPlainTests.connectionProperties());
+
+        PropertiesAsset appConfig = new PropertiesAsset()
+                        .addProperty(KafkaTestClientProvider.CONNECTION_PROPERTIES_KEY, KafkaTestClientProvider.encodeProperties(SaslPlainTests.connectionProperties()))
+                        .include(incomingProperties)
+                        .include(outgoingProperties)
+                        .include(connectorProperties);
+
+        RemoteFile bootstrapFile = server.getServerBootstrapPropertiesFile();
+        OutputStream bootstrapStream = bootstrapFile.openForWriting(true);
+        Properties props = appConfig.getFinalProperties();
+
+        createTwoConfigs(props, bootstrapStream); //This changes the contents of props
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addAsLibraries(kafkaClientLibs())
+                        .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                        .addPackage(KafkaSaslTestServlet.class.getPackage())
+                        .addPackage(BasicMessagingBean.class.getPackage())
+                        .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                        .addPackage(KafkaTestConstants.class.getPackage());
+        // This goes in bootstrap properties instead so checkpoint can update it
+        //                .addAsResource(appConfig, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+
+        KafkaUtils.copyTrustStore(SaslPlainTests.kafkaContainer, server);
+
+        server.setCheckpoint(CheckpointPhase.AFTER_APP_START, true,
+                             server -> {
+                                 assertNotNull("'SRVE0169I: Loading Web Module: " + APP_NAME + "' message not found in log before restore",
+                                               server.waitForStringInLogUsingMark("SRVE0169I: .*" + APP_NAME, 0));
+                                 assertNotNull("'CWWKZ0001I: Application " + APP_NAME + " started' message not found in log.",
+                                               server.waitForStringInLogUsingMark("CWWKZ0001I: .*" + APP_NAME + " started", 0));
+                                 try {
+                                     configureEnvVariable(server, ENV_AFTER_CHECKPOINT);
```

Comment:
Can you confirm that this is run after the checkpoint has been taken, just in case these variables are there during the checkpointing process

Reply:
This lamda is run after the checkpoint is taken.